src/bin/battery.rs

```diff
@@ -0,0 +1,244 @@
+        // read analog value from P0.28, P0.29 and P0.30
+        let saadc_result_0 = saadc.read_channel(&mut saadc_pin_0).unwrap() as f32;
+        let saadc_result_1 = saadc.read_channel(&mut saadc_pin_1).unwrap() as f32;
+        let saadc_result_2 = saadc.read_channel(&mut saadc_pin_2).unwrap() as f32;
+
+        // convert analog values to voltage 
+        let saadc_to_volt_0 = (saadc_result_0 * 0.6)/((1<<14) as f32 * (1.0/6.0));
+        let saadc_to_volt_1 = (saadc_result_1 * 0.6)/((1<<14) as f32 * (1.0/6.0));
+        let saadc_to_volt_2 = (saadc_result_2 * 0.6)/((1<<14) as f32 * (1.0/6.0));
+
+        // difference in voltage between P0.28 and P0.29
+        let temp_volt = (saadc_to_volt_1 - saadc_to_volt_2).abs();
+
+        // temperature = (-5.8151*current_voltage³ + 24.9708*current_voltage² - 61.8928*current_voltage + 48.8572)
+        let temp = -5.8151*(temp_volt*temp_volt*temp_volt) + 24.9708*(temp_volt*temp_volt) - 61.8928*temp_volt + 48.8572;
+        info!("The temperature is {}", temp);
+
+        // battery voltage Vm = Vb*10k/(150k+10k)
+        let bat_voltage = ((saadc_to_volt_0*(150000.0+10000.0))/10000.0)+1.3;
+
+        // battery voltage loss by temp function (approximated from us gov study: https://www.osti.gov/servlets/purl/975252) 
+        let bat_temp_ratio = 0.817 + 9.19*0.001*temp + -4.24*0.00001*temp*temp;
+
+        // voltage normailized for 25 degrees
+        let norm = bat_voltage/bat_temp_ratio;
+
+        // calculate the first state of charge
+        // State of charge estimation (linear estimation based on values form https://naturesgenerator.com/blogs/news/lead-acid-battery-voltage-chart?srsltid=AfmBOoqZi_WlVo8OX221jsoNeKqaDMWyDLfuOsrSwsqLOeA4DP-hocwW)
+        let old_soc = 0.251*norm -8.7;
+        info!("Old soc: {}", old_soc);
+
+
+        // set up CAN
+        // CAN PINS
+        let can_interrupt = port0.p0_24.into_pullup_input().degrade();
+        let cs0: Pin<Output<PushPull>> = port0.p0_20.into_push_pull_output(Level::High).degrade();
+        let sck: Pin<Output<PushPull>> = port0.p0_23.into_push_pull_output(Level::Low).degrade();
+        let miso: Pin<Input<Floating>> = port0.p0_21.into_floating_input().degrade();
+        let mosi: Pin<Output<PushPull>> = port0.p0_22.into_push_pull_output(Level::Low).degrade();
+
+        let spi_pins = hal::spi::Pins {
+            sck: Some(sck),
+            miso: Some(miso),
+            mosi: Some(mosi),
+        };
+
+        let spi_battery: Spi<SPI0> = Spi::new(
+            device.SPI0,
+            spi_pins,
+            Frequency::K250,
+            embedded_hal::spi::MODE_0,
+        );
+
+        // set up can setting
+        let settings = Mcp2515Settings::default();
+        let can_driver = Mcp2515Driver::init(spi_battery, cs0, can_interrupt, settings);
+
+        // new interrupt (should not be used as this firmware only sends data)
+        let gpiote = Gpiote::new(device.GPIOTE);
```

Comment:
GPIOT interrupt might be fully redundant. Can be left in place for completeness. If you want to remove it id recommend you run it by @wiiggee1 to make sure its alright since he knows how CAN works :)